### PR TITLE
Raise validation error if referring to unknown global or setting immutable global

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/parser_expr.cpp [hunk 1]
```diff
@@ -505,6 +505,19 @@ parser_result<Code> parse_expr(
         {
             uint32_t imm;
             std::tie(imm, pos) = leb128u_decode<uint32_t>(pos, end);
+
+            if (instr == Instr::global_get || instr == Instr::global_set)
+            {
+                if (module.globalsec.size() <= imm)
```
Comment (gumb0) [inline lib/fizzy/parser_expr.cpp:511]:
```suggestion
                if (module.get_global_count() <= imm)
```

File: lib/fizzy/types.hpp [hunk 2]
```diff
@@ -51,6 +51,9 @@ using TableIdx = uint32_t;
 // https://webassembly.github.io/spec/core/syntax/modules.html#syntax-memidx
```
Comment (gumb0) [inline lib/fizzy/types.hpp:51]: BTW these links now go to 1.1 spec

**Comment by gumb0** [comment]
Init expressions of globals need to be validated, too, but I would do that in another PR, as it seems a bit more complicated.

File: lib/fizzy/types.hpp (hunk 2 above)
Comment (axic) [reply lib/fizzy/types.hpp:61]: Do you want to open an issue/PR for us to direct-link to 1.0?

File: lib/fizzy/module.hpp [hunk 3]
```diff
@@ -74,5 +74,13 @@ struct Module
     {
         return !memorysec.empty() || !imported_memory_types.empty();
     }
+
+    bool is_global_mutable(GlobalIdx idx) const noexcept
```
Comment (axic) [inline lib/fizzy/module.hpp:78]: Hmm, shouldn't we have a unit test for this?

Comment (axic) [reply lib/fizzy/module.hpp:78]: For the sake of progress I'm merging this, but perhaps we should add unit tests for each module helper in a new PR>

File: lib/fizzy/types.hpp (hunk 2 above)
Comment (gumb0) [reply lib/fizzy/types.hpp:61]: https://github.com/wasmx/fizzy/issues/366

**Comment by gumb0** [comment]
Globals validation is poorly covered by spec tests, I added some cases to "to upstream" list

**Pull request merged**